packages/amazonq/src/extensionNode.ts

```diff
@@ -32,6 +32,11 @@ export async function activate(context: vscode.ExtensionContext) {
  * the code compatible with web and move it to {@link activateAmazonQCommon}.
  */
 async function activateAmazonQNode(context: vscode.ExtensionContext) {
+    // Do not run in tests since this corrupts the telemetry we try and assert
+    if (!env.isAutomation()) {
+        await (await CrashMonitoring.instance()).start()
```

Comment:
why is the isAutomation check needed? telemetry is already skipped in CI: https://github.com/aws/aws-toolkit-vscode/blob/56228d75eeeef5565f4e88ef156c605f6d15be4e/packages/core/src/shared/telemetry/telemetryClient.ts#L103-L105

If a test needs to ensure the metrics are reset before asserting them, then the test should just clear the metrics queue. That's already possible. Then that eliminates this isAutomation check (and the confusing comment above it).

Reply:
This was related to something else. But I found a way to handle this more cleanly in the tests.